File: app/src/main/java/com/joeshuff/dddungeongenerator/memory/MemoryGeneration.java
package com.joeshuff.dddungeongenerator.memory;

import androidx.annotation.NonNull;
import com.joeshuff.dddungeongenerator.generator.dungeon.Dungeon;
import com.joeshuff.dddungeongenerator.generator.dungeon.Modifier;

import java.text.SimpleDateFormat;
import java.util.Date;

public class MemoryGeneration implements Comparable<MemoryGeneration> {

    int epoch;
    private String generatedAt;

    private String seed;
    private int roomSize;
    private boolean longCorridors;
    private boolean loops = false;
    private double monsterFrequency = 1.0f;
    private double treasureFrequency = 1.0f;
    private double trapFrequency = 1.0f;
    private double stairsFrequency = 1.0f;
    private int depth = 50;

    public MemoryGeneration(String seed,
                            int roomSize,
                            boolean longCorridors,
                            double monsterFrequency,
                            double trapFrequency,
                            double treasureFrequency,
                            double stairsFrequency,
                            int depth) {

        SimpleDateFormat sdf = new SimpleDateFormat("dd MMM yyyy | HH:mm:ss");
        epoch = (int) (System.currentTimeMillis() / 1000d);
        generatedAt = sdf.format(new Date());

        this.seed = seed;
        this.roomSize = roomSize;
        this.longCorridors = longCorridors;
        this.monsterFrequency = monsterFrequency;
        this.trapFrequency = trapFrequency;
        this.treasureFrequency = treasureFrequency;
        this.stairsFrequency = stairsFrequency;
        this.depth = depth;
    }

    public String getGeneratedAt() {
        return generatedAt;
    }

    public String getSeed() {
        return seed;
    }

    public int getRoomSize() {
        return roomSize;
    }

    public boolean isLongCorridors() {
        return longCorridors;
    }

    public boolean isLoops() {
        return loops;
    }

    public Modifier getUserModifier() {
        return new Modifier()
                .setMonsterChanceModifier(monsterFrequency)
                .setTrapChanceModifier(trapFrequency)
                .setTreasureChanceModifier(treasureFrequency)
                .setStairChanceModifier(stairsFrequency)
                .setGrowthChanceUp(depth);
    }

    public double getMonsterFrequency() {
        return monsterFrequency;
    }

    public double getTreasureFrequency() {
        return treasureFrequency;
    }

    public double getTrapFrequency() {
        return trapFrequency;
    }

    public double getStairsFrequency() {
        return stairsFrequency;
    }

    public int getDepth() {
        return depth;
    }

    @Override
    public boolean equals(Object obj) {
        if (!(obj instanceof MemoryGeneration)) return false;
        return ((MemoryGeneration) obj).epoch == epoch;
    }

    @Override
    public int compareTo(@NonNull MemoryGeneration o) {
        int x = o.epoch;
        int y = epoch;

        return (x < y) ? -1 : ((x == y) ? 0 : 1);
    }

    public String workOutDungeonName() {
        Dungeon test = new Dungeon();
        test.setSeed(getSeed());

        test.generateAttributes();
        return test.getName();
    }
}

File: app/src/main/java/com/joeshuff/dddungeongenerator/memory/MemoryGeneration.kt
package com.joeshuff.dddungeongenerator.memory

import com.joeshuff.dddungeongenerator.generator.dungeon.Dungeon
import com.joeshuff.dddungeongenerator.generator.dungeon.Modifier
import java.text.SimpleDateFormat
import java.util.Date

class MemoryGeneration(val seed: String,
                       val roomSize: Int,
                       val longCorridors: Boolean,
                       val monsterFrequency: Double,
                       val trapFrequency: Double,
                       val treasureFrequency: Double,
                       val stairsFrequency: Double,
                       val depth: Int) : Comparable<MemoryGeneration> {
    var epoch: Int
    val generatedAt: String
    val isLoops = false

    val userModifier: Modifier
        get() = Modifier()
                .setMonsterChanceModifier(monsterFrequency)
                .setTrapChanceModifier(trapFrequency)
                .setTreasureChanceModifier(treasureFrequency)
                .setStairChanceModifier(stairsFrequency)
                .setGrowthChanceUp(depth)

    override fun equals(obj: Any?): Boolean {
        return if (obj !is MemoryGeneration) false else obj.epoch == epoch
    }

    override fun compareTo(o: MemoryGeneration): Int {
        val x = o.epoch
        val y = epoch
        return if (x < y) -1 else if (x == y) 0 else 1
    }

    fun workOutDungeonName(): String {
        val test = Dungeon()
        test.setSeed(seed)
        test.generateAttributes()
        return test.getName()
    }

    init {
        val sdf = SimpleDateFormat("dd MMM yyyy | HH:mm:ss")
        epoch = (System.currentTimeMillis() / 1000.0).toInt()
        generatedAt = sdf.format(Date())
    }
}